manifests/net_process_log.pp

```diff
@@ -0,0 +1,60 @@
+# @summary Logs user processes and open network connections to syslog
+#
+# Logs user processes and open network connections to syslog
+#
+# @param ignore_users
+#   String to pass to net_process_log.pl script as -i argument.
+#   Setting to empty string will use default defined in net_process_log.pl
+#
+# @param ps_arg
+#   String to pass to net_process_log.pl script as --psarg argument.
+#   Setting to empty string will use default defined in net_process_log.pl
+#
+# @param ss_arg
+#   String to pass to net_process_log.pl script as --ssarg argument.
+#   Setting to empty string will use default defined in net_process_log.pl
+#   
+# @param ss_filter
+#   String to pass to net_process_log.pl script as --ssfilter argument.
+#   Setting to empty string will use default defined in net_process_log.pl
+#
+# @param minute_interval
+#   Value to use in the minute field of the cron task 
+#
+# @example
+#   include profile_audit::net_process_log
+class profile_audit::net_process_log (
+  String $ignore_users,
+  String $ps_arg,
+  String $ss_arg,
+  String $ss_filter,
+  String $minute_interval,
+) {
+
+  file { '/root/scripts/net_process_log.pl':
+    ensure => 'present',
+    mode   => '0750',
+    owner  => 'root',
+    group  => 'root',
+    source => "puppet:///modules/${module_name}/net_process_log.pl",
+  }
+
+  # Setup any non-default options, otherwise *_option stays blank
+  # so we use default(s) from script
+  if $ignore_users != "" { $ignore_users_option = "-i \"${ignore_users}\"" }
+  if $ps_arg != "" { $ps_arg_option = "--psarg \"${ps_arg}\"" }
+  if $ss_arg != "" { $ss_arg_option = "--ssarg \"${ss_arg}\"" }
+  if $ss_filter != "" { $ss_filter_option = "--ssfilter \"${ss_filter}\"" }
+
+  cron { 'net_process_log':
+    user        => 'root',
+    minute      => $minute_interval,
+    hour        => '*',
+    month       => '*',
+    weekday     => '*',
+    monthday    => '*',
+    environment => ['SHELL=/bin/sh', ],
+    command     => "/root/scripts/net_process_log.pl ${ignore_users_option} ${ps_arg_option} ${ss_arg_option} ${ss_filter_option} >/dev/null 2>&1"
+  }
+
+}
```

Comment:
See comment about adding the `enable` logic into this subclass.

The other thing this then allows is to remove the CRONs, etc if not enabled.

Reply:
Module is now setup to clean up after itself if this logging is disabled after previously being enabled.